### PR TITLE
[ENH] add optional pretty printing with rich for logging

### PR DESCRIPTION
**Pull request by Remi-Gau**
- Closes none

Changes proposed in this pull request:

- use print from [rich](https://github.com/Textualize/rich) if users have it installed
- refactor tests for logger

This is just to try it out and see how the tests stdout look.

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Example of output

image

File: tox.ini [hunk 1]
```diff
@@ -100,6 +100,7 @@ description = run tests on latest version of all dependencies
 passenv = {[global_var]passenv}
 extras = test
 deps =
+    rich
```
Comment (Remi-Gau) [inline tox.ini:103]: I prefer to not add it as an "official" optional dependency in the pyproject.toml but prefer to still test that all test run fine if rich is installed.

**Review by bthirion — APPROVED** [review]
The logs indeed look good. Thx.

**Review by htwangtw — APPROVED** [review]
LGTM!

**Comment by Remi-Gau** [comment]
Thanks for the reviews @bthirion @htwangtw

**Pull request merged**